include/boost/iterator/detail/facade_iterator_category.hpp

```diff
@@ -114,9 +113,9 @@ struct iterator_facade_default_category
 // True iff T is convertible to an old-style iterator category.
 template <class T>
 struct is_iterator_category
-  : mpl::or_<
-        is_convertible<T,std::input_iterator_tag>
-      , is_convertible<T,std::output_iterator_tag>
+  : integral_constant<bool,
+      is_convertible<T,std::input_iterator_tag>::value ||
+      is_convertible<T,std::output_iterator_tag>::value
```

Comment:
This one isn't about static asserts, is it? If you are willing to convert conditions then this should probably be done in a separate PR and for all conditions, not just this one.

Also, this change is a potential regression in compile times as it always instantiates `is_convertible` twice. You can use `disjunction` from Boost.TypeTraits instead.

Also, `integral_constant` requires `#include <boost/type_traits/integral_constant.hpp>`, though it would be replaced with `#include <boost/type_traits/disjunction.hpp>` if you used `disjunction`.

Reply:
`mpl::or_` requires using `foo::type::value` instead of `foo::value` and this one was used in static asserts hence the change to avoid the inconsistency. I omitted the include as availability of `is_convertible` implies `integral_constant`

Anyway to keep this as small as possible I reverted this change, it's still in the history for the future if needed.

Reply:
> I omitted the include as availability of `is_convertible` implies `integral_constant`

I assume no implicit includes. If the code uses a component directly, its header must be included.

> Anyway to keep this as small as possible I reverted this change

Ok, thanks.